### ISSUE TITLE
Crash when changing time signature

### ISSUE THREAD

**Issue by jonarnoldmusic** [musescore/MuseScore#18021]
### Issue type

Crash or freeze

### Bug description

When adding a new time signature in any measure of the attached score, Musescore 4 crashes.

### Steps to reproduce

Add a time signature change in any measure

### MuseScore Version

OS: Windows 10 Version 2009, Arch.: x86_64, MuseScore version (64-bit): 4.0.2-230651553, revision: github-musescore-musescore-dbe7c6d

Also in master (e.g. 4.1.0-231680305, revision: github-musescore-musescore-b79c97b)

### Regression

Yes, this used to work in a previous version of MuseScore 4.x

### Operating system

Windows 11

### Additional context

[musescoreDiag.zip](https://github.com/musescore/MuseScore/files/11773252/musescoreDiag.zip)
[msczFile.zip](https://github.com/musescore/MuseScore/files/11773261/msczFile.zip)

**Comment by cbjeukendrup** [comment]
The problem is that the Organ part score is entirely corrupted (it is simply empty). You can fix it by going to the Parts dialog, clicking the '...' button for the Organ part, and clicking "Reset". Then it will work again. 

The big question is of course: how did this score get into this state? The score file itself doesn't really give us a clue, so if you remember something about what you've done with this Organ part, that could be very helpful.

**Comment by cbjeukendrup** [comment]
I looked at the log files. Some of them contain the following error message:
```
MasterScore | rebuildExcerptsMidiMapping: rebuildExcerptsMidiMapping: no part in master score is linked with Organ
```

This _must_ be related to the inconsistent state of the file. It looks like this message is seem for the first time in the log file called `MuseScore_230616_140139.log`, so probably that is when the problem was introduced.

In this file, I can see you did the following: (keep in mind that between these things, you might have done other things that are not included in the log file)
1. You did something with Staff Properties
2. You opened the "Add/Remove instruments" dialog but canceled
3. You opened the "Add/Remove instruments" dialog two more times, and did not cancel, but I can't see what you did inside it
4. You deleted some note or other notation element
5. You saved the file
6. You opened the Parts dialog
8. You went to Staff Properties again
9. You opened the Parts dialog again
10. You saved the file again
11. Parts dialog again
12. Save again
13. And this time, most likely while saving, the error messages are printed for the first time.
14. You opened the parts dialog once more, but didn't seem to make any edits. After that, MuseScore was quit.

I know it's an impossible question, but do you by any chance remember what you did in the Parts dialog, especially that last time before the error message appeared?

**Comment by jonarnoldmusic** [comment]
Hi Casper-

I don't know all the exact steps or sequence I did. I do know that I renamed several of the instruments and tried to hide some of the staves of the organ part. (I only wanted a bass staff for continuo.) When I couldn't hide individual staves of the organ part, I deleted it (I believe with the trash can in the Instruments panel) and added a cello part. As you can see, the cello part did not appear in the Parts dialog, but the organ part was still retained in the Parts dialog and ended up empty.

Is that helpful?

Jon

**Comment by cbjeukendrup** [comment]
@jonarnoldmusic Thanks! I investigated this issue again today, and came to the conclusion that it is most likely that you did approximately the following:
1. Open Parts dialog -> a list of available part scores is generated
2. You remove the organ and add the cello in its place (and thus the cello takes the instrument ID of the organ, because that ID is free again after removing the organ)
3. Open the Parts dialog again -> the list of available part scores is not updated correctly, because the list of used instrument IDs is the same as before. You open the Organ part, but this part is in an inconsistent state, because it still tries to refer to the Organ instrument but that has been removed. 
4. Save close reopen
5. Now, adding a time signature anywhere does indeed cause a crash. 

Fortunately, this way of producing a broken score (i.e. the error that occurred in step 3) was reported in issue #17395, which has been fixed by #17408. That means that we can consider this issue completed!

Bonus tip: it _is_ possible to hide or delete individual staves of instruments, namely by clicking the little triangle button in the Instruments panel:

https://github.com/musescore/MuseScore/assets/48658420/8a95dda0-d120-456d-986a-56e3ad2b165c
